tests/PHPStan/Analyser/nsrt/get-defined-vars.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace GetDefinedVars;
+
+use function PHPStan\Testing\assertType;
+
+function doFoo(int $param) {
```

Comment:
It should be handled by PHPStan but I will add tests for that too. On global level it would be just array<string, mixed> because any variable can exists in global scope. Inside functions it can be determined form use og `$global $var` so it should be resolvable too.